src/API/Middlewares/HttpExceptionsMiddleware.test.ts

```diff
@@ -0,0 +1,114 @@
+import { IllegalArgumentException, NotFoundException } from '@po.et/poet-js'
+import { describe } from 'riteway'
+import { spy } from 'sinon'
+import { getCodeError, HttpExceptionsMiddleware, executeLogger } from './HttpExceptionsMiddleware'
+
+describe('HttpExceptionsMiddleware middleware', async (should: any) => {
+  const { assert } = should('')
+
+  {
+    const ctx = {}
+    const next = spy()
+    await HttpExceptionsMiddleware(ctx, next)
+
+    assert({
+      given: 'no errors',
+      should: 'call next once',
+      actual: next.calledOnce,
+      expected: true,
+    })
+  }
+
+  {
+    const ctx = {}
+    const next = spy()
+    await HttpExceptionsMiddleware(ctx, next)
+
+    const actual = ctx
+    const expected = ctx
+
+    assert({
+      given: 'no errors',
+      should: 'the ctx keep without changes',
+      actual,
+      expected,
+    })
+  }
+
+  {
+    const ctx = { status: 0, body: '' }
+    const next = () => Promise.reject(new Error('Something happened!'))
+
+    await HttpExceptionsMiddleware(ctx, next).catch()
+
+    const actual = { status: ctx.status, body: ctx.body }
+    const expected = { status: 503, body: 'Something happened!' }
+
+    assert({
+      given: 'an error',
+      should: 'return context with status 503 and the text Something happened!',
+      actual,
+      expected,
+    })
+  }
+})
+
+describe('HttpExceptionsMiddleware getCodeError()', async (should: any) => {
+  const { assert } = should('')
+
+  {
+    const illegalArgumentException = new IllegalArgumentException('IllegalArgumentException')
+    const actual = getCodeError(illegalArgumentException)
+    const expected = { status: 422, body: 'IllegalArgumentException' }
+
+    assert({
+      given: 'a IllegalArgumentException error',
+      should: 'return status with 422 and body with IllegalArgumentException',
+      actual,
+      expected,
+    })
+  }
+
+  {
+    const notFoundException = new NotFoundException('NotFoundException')
+    const actual = getCodeError(notFoundException)
+    const expected = { status: 404, body: 'NotFoundException' }
+
+    assert({
+      given: 'a NotFoundException error',
+      should: 'return status with 404 and body with NotFoundException',
+      actual,
+      expected,
+    })
+  }
+
+  {
+    const actual = getCodeError(new Error('Error'))
+    const expected = { status: 503, body: 'Error' }
+
+    assert({
+      given: 'any kind of Error distinct to IllegalArgumentException or NotFoundException',
+      should: 'return status with 503 and body with Error',
+      actual,
+      expected,
+    })
+  }
+})
+
+describe('HttpExceptionsMiddleware executeLogger()', async (should: any) => {
+  const { assert } = should('')
+  {
+    const context = { logger: { error: spy() } }
+    const exception = new Error()
+    const status = 200
+
+    executeLogger(context, exception, status)
```

Comment:
We can avoid testing this side effect (calling the logger) and increase readability a bit by introducing another mini refactor. See related comment.

Reply:
I need to test the method `executeLogger ` if not I get this result 

```
src/API/Middlewares               |     92.5 |    64.29 |    90.91 |    93.94 |                   |
  HttpExceptionsMiddleware.ts     |    92.59 |    33.33 |    88.89 |    95.24 |                20 |
```

instead of:

```
src/API/Middlewares               |     97.5 |    92.86 |      100 |    96.97 |                   |
  HttpExceptionsMiddleware.ts     |      100 |      100 |      100 |      100 |                   |
```